### PR TITLE
Highlight

### PR DESCRIPTION
**Pull request by Yomguithereal**
Ok @glenjamin, here is a first draft for syntax highlight so you see how this solution works (note that it does fix the react warning, as a nice collateral benefit).

Don't merge this yet, here is what I must add before completing the PR:

- [ ] Find a nicer looking stylesheet for highlight (the one I got is ugly).
- [x] Check that JSX is correctly handled.
- [ ] Applying this solution to JS datatypes display.
- [ ] Add some unit tests?

### CODE REVIEW

File: lib/code-renderer.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+module.exports = function codeRenderer(node, props, children) {
+  var lang = (node.info || '').trim() || null;
+  var className = 'language-' + lang;
+
+  var highlightedCode = highlight(lang, children[0]);
+
+  var codeElement = $('code', {
+    className: className,
+    dangerouslySetInnerHTML: {
```
Comment (glenjamin) [inline lib/code-renderer.js]: I'm not sure about this, if we just spat out a `<code>` and then `hljs.highlightBlock(ref);` there'd be no need to `dangerouslySetInnerHTML`.

File: lib/code-renderer.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+var React = require('react');
+var $ = React.createElement;
+
+var highlight = require('./highlight');
+
+module.exports = function codeRenderer(node, props, children) {
+  var lang = (node.info || '').trim() || null;
+  var className = 'language-' + lang;
```
Comment (glenjamin) [inline lib/code-renderer.js]: if you include a unique string in this `className`, we can then scope all the hljs styles to only apply to the code blocks

`devcards-syntax-highlighted-block` or similar.

File: lib/code-renderer.js (hunk 1 above)
Comment (Yomguithereal) [inline lib/code-renderer.js]: Yes but we have to use lifecycle methods, and we assume we are in the DOM. This works if you want to render server-side for instance.

Comment (glenjamin) [inline lib/code-renderer.js]: I'm not sure server rendering is a core use-case for devcards.

Even with server rendering, you need JS enabled to use devcards properly - and the `mount` lifecycle would still fire and enable the highlighting.

I might have a go this evening at what it'd look like with lifecycle hooks - then we can compare

**Comment by Yomguithereal** [comment]
I'll let this PR be the time you do tests with lifecycle if you want. You can also merge it if you want to reuse the logic parts. We'll deal with JS datatypes display when this is solved.

**Comment by glenjamin** [comment]
I've just pushed https://github.com/glenjamin/devcards-js/tree/highlight-with-highlight - i'm pretty happy with that approach, minimal code and it highlight plugs in white quite neatly.

I'm not massively happy with highlight JS's actual highlighting though - all the existing code samples don't actually get any parsing applied.

I'm going to have a go with http://prismjs.com/

**Comment by glenjamin** [comment]
And here's prism: https://github.com/glenjamin/devcards-js/tree/highlight-with-prism

**Comment by glenjamin** [comment]
I reckon #23 is my preferred approach so far

**Comment by Yomguithereal** [comment]
Yes, #23 looks nice.

**Pull request closed without merging**